cartographer_ros/cartographer_ros/assets_writer.cc

```diff
@@ -148,18 +143,51 @@ std::unique_ptr<carto::io::PointsBatch> HandleMessage(
   return points_batch;
 }
 
-void RunPipeline(
-    const std::vector<std::unique_ptr<carto::io::PointsProcessor>>& pipeline,
-    const std::vector<std::string>& bag_filenames,
-    const carto::mapping::proto::PoseGraph& pose_graph_proto,
-    const std::string tracking_frame, const std::string& urdf_filename,
-    const bool use_bag_transforms) {
+}  // namespace
+
+AssetsWriter::AssetsWriter(const std::string& pose_graph_filename,
+                           const std::vector<std::string>& bag_filenames,
+                           const std::string& output_file_prefix)
+    : bag_filenames_(bag_filenames),
+      pose_graph_(LoadPoseGraph(pose_graph_filename)) {
+  CHECK_EQ(pose_graph_.trajectory_size(), bag_filenames_.size())
+      << "Pose graphs contains " << pose_graph_.trajectory_size()
+      << " trajectories while " << bag_filenames_.size()
+      << " bags were provided. This tool requires one bag for each "
+         "trajectory in the same order as the correponding trajectories in the "
+         "pose graph proto.";
+
+  // This vector must outlive the pipeline.
+  all_trajectories_.resize(pose_graph_.trajectory_size());
+  std::copy(pose_graph_.trajectory().begin(), pose_graph_.trajectory().end(),
```

Comment:
Did you consider `all_trajectories_ = std::vector<...>(...begin(), ...end())`? Seems more readable to me.

Reply:
Done